### PR TITLE
[PropertyAccess] Document usage of createPropertyAccessor args

### PR DESCRIPTION
**Pull request by chalasr**
This documents https://github.com/symfony/symfony/pull/18977 (on hold).
Suggestions to make it better would be really appreciated.

### CODE REVIEW

**Comment by chalasr** [comment]
Not sure that the failed test on travis is related.

File: components/property_access/introduction.rst [hunk 1]
```diff
@@ -46,7 +46,12 @@ method. This is done using the index notation that is used in PHP::
     var_dump($accessor->getValue($person, '[first_name]')); // 'Wouter'
     var_dump($accessor->getValue($person, '[age]')); // null
 
-As you can see, the method will return ``null`` if the index does not exists.
+.. caution::
+
+    As you can see, the method will return ``null`` if the index does not exists.
+    To make it throws an exception in case of invalid index, set the first argument
```
Comment (ogizanagi) [inline components/property_access/introduction.rst]:
```diff
-To make it throws
+ To make it throws
```

(present infinitive)

File: components/property_access/introduction.rst [hunk 2]
```diff
@@ -321,6 +326,13 @@ instead::
         // ...
     }
 
+.. caution::
+
+    Calling :method:`PropertyAccess::createPropertyAccessor<Symfony\\Component\\PropertyAccess\\PropertyAccess::createPropertyAccessor()>`
+    with an array for an invalid index would always return ``true``. To make it returns ``false``, set the first argument
```
Comment (ogizanagi) [inline components/property_access/introduction.rst]:
```diff
- To make it returns ``false``
+ To make it return ``false``
```

File: components/property_access/introduction.rst [hunk 3]
```diff
@@ -46,7 +46,12 @@ method. This is done using the index notation that is used in PHP::
     var_dump($accessor->getValue($person, '[first_name]')); // 'Wouter'
     var_dump($accessor->getValue($person, '[age]')); // null
 
-As you can see, the method will return ``null`` if the index does not exists.
+.. caution::
+
+    As you can see, the method will return ``null`` if the index does not exists.
+    To make it throw an exception in case of invalid index, set the first argument
+    :method:`PropertyAccess::createPropertyAccessor<Symfony\\Component\\PropertyAccess\\PropertyAccess::createPropertyAccessor()>`
+    to true, e.g. ``PropertyAccess::createPropertyAccessor(true)``.
```
Comment (OskarStark) [inline components/property_access/introduction.rst]: true in ticks? not sure

File: components/property_access/introduction.rst [hunk 4]
```diff
@@ -46,7 +46,12 @@ method. This is done using the index notation that is used in PHP::
     var_dump($accessor->getValue($person, '[first_name]')); // 'Wouter'
     var_dump($accessor->getValue($person, '[age]')); // null
 
-As you can see, the method will return ``null`` if the index does not exists.
+.. caution::
+
+    As you can see, the method will return ``null`` if the index does not exists.
+    To make it throw an exception in case of invalid index, set the first argument
```
Comment (ogizanagi) [inline components/property_access/introduction.rst]:
```diff
- set the first argument
+ set the first argument of
```

?

File: components/property_access/introduction.rst [hunk 5]
```diff
@@ -321,6 +326,13 @@ instead::
         // ...
     }
 
+.. caution::
+
+    Calling :method:`PropertyAccess::createPropertyAccessor<Symfony\\Component\\PropertyAccess\\PropertyAccess::createPropertyAccessor()>`
+    with an array for an invalid index would always return ``true``. To make it return ``false``, set the first argument
```
Comment (ogizanagi) [inline components/property_access/introduction.rst]: same here

File: components/property_access/introduction.rst (hunk 4 above)
Comment (chalasr) [inline components/property_access/introduction.rst]: Totally, a typo

File: components/property_access/introduction.rst (hunk 3 above)
Comment (chalasr) [inline components/property_access/introduction.rst]: Do you mean that I should wrap it in ` ```` ` chars? It's a mistake that I'll correct, thank's for catch it.

Comment (OskarStark) [inline components/property_access/introduction.rst]: thats what i meant, thank you!

**Comment by chalasr** [comment]
The related PR has been merged (https://github.com/symfony/symfony/pull/18977).
It also introduced a second argument corresponding to `PropertyAccessorBuilder::$magicCall`, let me know if you think that I should also adapt the documented usage of this argument (actually documented through `PropertyAccessorBuilder::enableMagicCall`, not directly from `PropertyAccess::createPropertyAccessor()` that can now be used).

**Comment by chalasr** [comment]
Updated according to https://github.com/symfony/symfony/pull/19040.
Please don't merge it before.

**Comment by fabpot** [comment]
Closing as #18977 has been reverted.

**Pull request closed without merging**